packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -65,10 +65,11 @@ ComponentNodeManager.create = function(renderNode, env, options) {
   // Instantiate the component
   component = createComponent(component, isAngleBracket, createOptions, renderNode, env, attrs);
 
-  // If the component specifies its template via the `layout properties
-  // instead of using the template looked up in the container, get them
+  // If the component specifies its layout via the `layout` property
+  // instead of using the template looked up in the container, get it
   // now that we have the component instance.
   layout = get(component, 'layout') || layout;
+  component._hasBlock = !!templates.default;
```

Comment:
This should be provided as an option in `createOptions` (so that it is available at `init` and doesn't change the shape of the object after initialization).